go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -244,3 +253,72 @@ func runOvnKube(ctx *cli.Context) error {
 
 	return fmt.Errorf("need to run ovnkube in either master and/or node mode")
 }
+
+// watchForChanges closes stopCh if the configuration file changed.
+func watchForChanges(configPath string, stopCh chan struct{}) error {
+	if configPath == "" {
+		return nil
+	}
+	configPath, err := filepath.Abs(configPath)
+	if err != nil {
+		return err
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	defer watcher.Close()
+
+	go func() {
+		for {
+			select {
+			case <-stopCh:
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					logrus.Infof("Configuration file %s changed, exiting...", event.Name)
+					close(stopCh)
+					os.Exit(0)
+					return
+				}
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				logrus.Errorf("fsnotify error %v", err)
+			}
+		}
+	}()
+
+	done := make(chan bool)
+	// Watch all symlinks for changes
+	p := configPath
+	maxdepth := 100
+	for depth := 0; depth < maxdepth; depth++ {
+		if err := watcher.Add(p); err != nil {
+			return err
+		}
+		logrus.Infof("Watching config file %s for changes", p)
+
+		stat, err := os.Lstat(p)
+		if err != nil {
+			return err
+		}
+
+		// configmaps are usually symlinks
+		if stat.Mode()&os.ModeSymlink > 0 {
+			p, err = filepath.EvalSymlinks(p)
+			if err != nil {
+				return err
+			}
+		} else {
+			break
+		}
+	}
+	<-done
+
```

Comment:
L321 is a `blocking` statement. So, the caller runOvnKube() is blocked forever. Did you mean to run watchForChanges() as a go routine?